plugins/postcss-is-pseudo/src/split-selectors/split-selectors.ts

```diff
@@ -0,0 +1,175 @@
+
+			const replacementsParts = [];
+
+			pseudo.nodes.forEach((child) => {
+				const replacement = {
+					start: start,
+					end: end,
+					option: '',
+				};
+
+				const childSpecificity = selectorSpecificity(child);
+				let childCSS = child.toString().trim();
+
+				const aSpecificityCorrection = Math.max(0, specificity.a - childSpecificity.a);
+				const bSpecificityCorrection = Math.max(0, specificity.b - childSpecificity.b);
+				const cSpecificityCorrection = Math.max(0, specificity.c - childSpecificity.c);
+
+				for (let i = 0; i < aSpecificityCorrection; i++) {
+					childCSS += doesNotExistId;
+				}
+				for (let i = 0; i < bSpecificityCorrection; i++) {
+					childCSS += doesNotExistClass;
+				}
+				for (let i = 0; i < cSpecificityCorrection; i++) {
+					childCSS += doesNotExistTag;
+				}
+
+				replacement.option = childCSS;
+				replacementsParts.push(replacement);
+			});
+
+			replacements.push(replacementsParts);
+		});
+
+		if (!replacements.length) {
+			return [selector];
+		}
+
+		const results = [];
+		cartesianProduct(...replacements).forEach((replacement) => {
+			let result = '';
+
+			for (let i = 0; i < replacement.length; i++) {
+				const options = replacement[i];
+
+				result += selector.substring(replacement[i - 1]?.end || 0, replacement[i].start);
+				result += ':is(' + options.option + ')';
+
+				if (i === replacement.length - 1) {
+					result += selector.substring(replacement[i].end);
+				}
+			}
+
+			results.push(result);
+		});
+
+		let formattedResults = results.map((x) => {
+			const modifiedSelectorAST = parser().astSync(x);
+
+			// Handle complex selector cases
+			modifiedSelectorAST.walk((node) => {
+				childAdjacentChild(node);
+			});
+
+			// Remove `:is` with single elements
+			modifiedSelectorAST.walkPseudos((pseudo) => {
+				if (pseudo.value !== ':is' || !pseudo.nodes || pseudo.nodes.length !== 1) {
+					return;
+				}
+
+				// Warn when `:is` contains a complex selector.
+				pseudo.nodes.forEach((child) => {
+					if (child.some((node) => node.type === 'combinator')) {
+						warnFn();
+					}
+				});
+
+				pseudo.replaceWith(pseudo.nodes[0]);
+			});
+
+			modifiedSelectorAST.walk((node) => {
+				if ('nodes' in node) {
+					node.nodes.forEach((child) => {
+						sortCompoundSelectorsInsideComplexSelector(child);
+					});
+					sortCompoundSelectorsInsideComplexSelector(node);
+				}
+			});
+
+			return modifiedSelectorAST.toString();
+		});
+
+		if (foundNestedIs) {
+			formattedResults = splitSelectors(formattedResults, pluginOptions, warnFn);
+		}
+
+		return formattedResults;
+	}).filter((x) => {
+		return !!x;
+	});
+}
+
+// https://en.wikipedia.org/wiki/Cartesian_product
+function cartesianProduct(...args) {
+	const r = [], max = args.length - 1;
+	function helper(arr, i) {
+		for (let j = 0, l = args[i].length; j < l; j++) {
+			const a = arr.slice(0); // clone arr
+			a.push(args[i][j]);
+			if (i == max)
+				r.push(a);
+			else
+				helper(a, i + 1);
```

Comment:
[nit] Can we be consistent with the brackets?

Reply:
good catch! I'll check if we can make the linter fix this.
Code was copy/paste

Reply:
I have added eslint rules for tabs and braces.
This should no longer pass CI.